src/coreclr/gc/gc.cpp

```diff
@@ -8389,29 +8404,260 @@ void destroy_card_table (uint32_t* c_table)
     dprintf (2, ("Table Virtual Free : %Ix", (size_t)&card_table_refcount(c_table)));
 }
 
+void gc_heap::get_card_table_element_sizes (uint8_t* start, uint8_t* end, size_t sizes[total_bookkeeping_elements])
+{
+    memset (sizes, 0, sizeof(size_t) * total_bookkeeping_elements);
+    sizes[card_table_element] = size_card_of (start, end);
+    sizes[brick_table_element] = size_brick_of (start, end);
+#ifdef CARD_BUNDLE
+    if (can_use_write_watch_for_card_table())
+    {
+        sizes[card_bundle_table_element] = size_card_bundle_of (start, end);
+    }
+#endif //CARD_BUNDLE
+#ifdef FEATURE_USE_SOFTWARE_WRITE_WATCH_FOR_GC_HEAP
+    if (gc_can_use_concurrent)
+    {
+        sizes[software_write_watch_table_element] = SoftwareWriteWatch::GetTableByteSize(start, end);
+    }
+#endif // FEATURE_USE_SOFTWARE_WRITE_WATCH_FOR_GC_HEAP
+    sizes[seg_mapping_table_element] = size_seg_mapping_table_of (start, end);
+#ifdef BACKGROUND_GC
+    if (gc_can_use_concurrent)
+    {
+        sizes[mark_array_element] = size_mark_array_of (start, end);
+    } 
+#endif //BACKGROUND_GC
+}
+
+void gc_heap::get_card_table_element_layout (uint8_t* start, uint8_t* end, size_t layout[total_bookkeeping_elements + 1])
+{
+    size_t sizes[total_bookkeeping_elements];
+    get_card_table_element_sizes(start, end, sizes);
+
+    size_t alignment[total_bookkeeping_elements + 1];
+    alignment[card_table_element] = 1;
+    alignment[brick_table_element] = 2;
+#ifdef CARD_BUNDLE
+    alignment[card_bundle_table_element] = 1;
```

Comment:
it'd be great to not use these hard coded numbers, eg, you could replace 2 with size_of (short).

Reply:
also I think it should not be 1 for card_table/card_bundle_table - they should all be aligned at least on their natural word size which means 

card_table - 4 bytes
brick_table - 2 bytes
card_bundle_table - 4 bytes